cmd/node/node.go

```diff
@@ -960,7 +973,10 @@ func (app *App) LoadOrCreateEdSigner() (*signing.EdSigner, error) {
 		log.With().Info("created new identity", edSgn.PublicKey())
 		return edSgn, nil
 	}
-	edSgn, err := signing.NewEdSignerFromKey(data, signing.WithPrefix(app.Config.Genesis.GenesisID().Bytes()))
+	edSgn, err := signing.NewEdSigner(
+		signing.WithPrivateKey(data),
```

Comment:
it's strange that private key is passed as option. shouldn't it be a required field?

Reply:
this is the only place in the whole codebase where `WithPrivateKey` is used. Everywhere else the signer is instantiated with `signing.NewEdSigner()` and will generate a random key.

Before there were 2 functions to create a signer: `NewEdSigner` and `NewEdSignerFromKey`, the later one has been replaced with this functional option.